README.md

```diff
@@ -1,1329 +1,100 @@
+It is presented as short tutorial series.
 
 ```bash
-1_hello_world/
-
-├── README.md
-├── package.json
-├── tsconfig.json
-├── src
-│   └── index.ts
-└── style
-    └── index.css
+git clone https://github.com/jtpio/jupyterlab-extension-tutorial.git && 
+  cd jupyterlab-extension-tutorial && \
+  conda env create -f environment.yml && \
+  conda activate jupyterlab-extension-tutorial && \
+  cd hello_world && \
+  npm install && \
+  npm run build && \
+  jupyter labextension link .
+# In another terminal
+jupyter lab --watch
+# open http:/localhost:8888?token=???
 ```
 
-* `README.md` contains some instructions
-* `package.json` contains information about the extension such as dependencies
-* `tsconfig.json` contains information for the typescript compilation
-* `src/index.ts` _this contains the actual code of our extension_
-* `style/index.css` contains style elements that we can use
+## Develop by Examples
 
-What does this extension do? You don't need a PhD in Computer Science to take a
-guess from the title of this section, but let's have a closer look:
+You may find easier to learn how to extend `by examples` instead of going through the [RTFM](https://en.wikipedia.org/wiki/RTFM). Start with the `hello_world` and jump then to topic you are interested into.
 
-#### A minimal extension that prints to the browser console ####
++ [Hello World](./hello_world)
++ [Commands and Menus](./commands_and_menus)
++ [Signals and Buttons](./signals_and_buttons)
++ [Widgets](./widgets)
++ [Datagrid](./datagrid) 
++ [Jupyter Widgets](./jupyter_widgets)
++ [Kernel Output](./kernel_output)
++ [Kernel Messaging](./kernel_messaging)
++ [Serving Files](./serving_files)
 
-We start with the file `src/index.ts`. This typescript file contains the main
-logic of the extension. It begins with the following import section:
+## About JupyterLab
 
-```typescript
-import {
-  JupyterLab, JupyterLabPlugin
-} from '@JupyterLab/application';
+Jupyterlab can be used as a platform to combine existing data-science components into a 
+new powerful application that can be deployed remotely for many users. Some of the higher 
+level components that can be used are text editors, terminals, notebooks, interactive widgets, 
+filebrowser, renderers for different file formats that provide access to an enormous ecosystem 
+of libraries from different languages.
 
-import '../style/index.css';
-```
+## Prerequisites
 
-`JupyterLab` is class of the main Jupyterlab application. It allows us to
-access and modify some of its main components. `JupyterLabPlugin` is the class
-of the extension that we are building. Both classes are imported from a package
-called `@JupyterLab/application`. The dependency of our extension on this
-package is declared in the file `package.json`:
-
-```json
-  "dependencies": {
-    "@JupyterLab/application": "^0.16.0"
-  },
-```
+Writing an extension is not particularly difficult but requires very basic knowledge of javascript 
```

Comment:
"javascript" -> "JavaScript` and "Typescript" -> "TypeScript"? So it's also consistent with the JupyterLab documentation.